packages/abi-typegen/src/templates/utils/formatEnums.ts

```diff
@@ -0,0 +1,24 @@
+import type { EnumType } from '../../abi/types/EnumType';
+import { TargetEnum } from '../../types/enums/TargetEnum';
+import type { IType } from '../../types/interfaces/IType';
+
+export function formatEnums(params: { types: IType[] }) {
+  const { types } = params;
+
+  const enums = types
+    .filter((t) => t.name === 'enum')
+    .map((t) => {
+      const et = t as EnumType; // only enums here
+      const structName = et.getStructName();
+      const inputValues = et.getStructContents({ types, target: TargetEnum.INPUT });
+      const outputValues = et.getStructContents({ types, target: TargetEnum.OUTPUT });
+      return {
+        structName,
+        inputValues,
+        outputValues,
+        recycleRef: inputValues === outputValues, // reduces duplication
+      };
+    });
+
+  return { enums };
```

Comment:
That makes sense; I'll give it a final look.

Reply:
> Seems like a bunch of similar code in the formatters, at least here and in the struct one.

So, I always strive for DRY implementations, but sometimes it requires adding a layer of _abstraction_ that makes code harder _(not easier)_ to maintain. After giving it some thought, I believe this would likely be the case here.

> Should we move formatting into the class itself?

No, because the class you're referring to _is_ the `*Type` class, whereas the formatters in question address things on a higher level _(for the `templates`)_, thus dealing with multiple _`*Types`_.